components/registry-facade/pkg/registry/registry.go

```diff
@@ -170,12 +176,51 @@ func NewRegistry(cfg config.Config, newResolver ResolverProvider, reg prometheus
 		}
 		specProvider[api.ProviderPrefixRemote] = specprov
 	}
+	if cfg.FixedSpecProvider != "" {
+		fc, err := ioutil.ReadFile(cfg.FixedSpecProvider)
+		if err != nil {
+			return nil, xerrors.Errorf("cannot read fixed spec: %w", err)
+		}
+
+		var fp api.ImageSpec
+		err = jsonpb.UnmarshalString(string(fc), &fp)
+		if err != nil {
+			return nil, xerrors.Errorf("cannot unmarshal fixed spec: %w", err)
+		}
+		specProvider[api.ProviderPrefixFixed] = FixedImageSpecProvider(fp)
+	}
+
+	var ipfs *IPFSStore
+	if cfg.IPFSCache != nil && cfg.IPFSCache.Enabled {
+		addr := cfg.IPFSCache.IPFSAddr
+		if ipfsHost := os.Getenv("IPFS_HOST"); ipfsHost != "" {
+			addr = strings.ReplaceAll(addr, "$IPFS_HOST", ipfsHost)
+		}
+
+		maddr, err := ma.NewMultiaddr(strings.TrimSpace(addr))
+		if err != nil {
+			return nil, xerrors.Errorf("cannot connect to IPFS: %w", err)
+		}
+		core, err := httpapi.NewApi(maddr)
+		if err != nil {
+			return nil, xerrors.Errorf("cannot connect to IPFS: %w", err)
+		}
+		rdc := redis.NewClient(&redis.Options{
```

Comment:
Please change it to `redis.NewSentinelClient`. If we commit to this, a single Redis instance is a no go from the start.
Also, please add Username and Password (same reason)